### PR TITLE
Fix build on Fedora rawhide and add Dockerfile for building on Fedora 28.

### PR DESCRIPTION
**Pull request by adelton**

### CODE REVIEW

File: Dockerfile.fedora-rawhide [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
 RUN groupadd -g 288 kdcproxy ; useradd -u 288 -g 288 -c 'IPA KDC Proxy User' -d '/var/lib/kdcproxy' -s '/sbin/nologin' kdcproxy
 RUN groupadd -g 289 ipaapi; useradd -u 289 -g 289 -c 'IPA Framework User' -d / -s '/sbin/nologin' ipaapi
 
-RUN mkdir -p /run/lock && dnf upgrade -y && dnf install -y freeipa-server freeipa-server-dns freeipa-server-trust-ad initscripts && dnf clean all
+RUN dnf upgrade --disablerepo=rawhide-modular -y && dnf install -y freeipa-server freeipa-server-dns freeipa-server-trust-ad initscripts && dnf clean all
 
 # Workaround 1364139
-RUN sed -i '/installutils.verify_fqdn(config.master_host_name, options.no_host_dns)/s/)/, local_hostname=False)/' /usr/lib/python2.7/site-packages/ipaserver/install/server/replicainstall.py && python -m compileall /usr/lib/python2.7/site-packages/ipaserver/install/server/replicainstall.py
+RUN sed -i '/installutils.verify_fqdn(config.master_host_name, options.no_host_dns)/s/)/, local_hostname=False)/' /usr/lib/python3.6/site-packages/ipaserver/install/server/replicainstall.py && python -m compileall /usr/lib/python3.6/site-packages/ipaserver/install/server/replicainstall.py
```
Comment (stlaz) [inline Dockerfile.fedora-rawhide]: See the line right below this one :)

File: Dockerfile.fedora-rawhide [hunk 2]
```diff
@@ -1,15 +1,15 @@
 # Clone from the Fedora rawhide image
 FROM registry.fedoraproject.org/fedora:rawhide
 
-MAINTAINER Jan Pazdziora
+MAINTAINER FreeIPA Developers <freeipa-devel@lists.fedorahosted.org>
 
 RUN groupadd -g 288 kdcproxy ; useradd -u 288 -g 288 -c 'IPA KDC Proxy User' -d '/var/lib/kdcproxy' -s '/sbin/nologin' kdcproxy
 RUN groupadd -g 289 ipaapi; useradd -u 289 -g 289 -c 'IPA Framework User' -d / -s '/sbin/nologin' ipaapi
 
-RUN mkdir -p /run/lock && dnf upgrade -y && dnf install -y freeipa-server freeipa-server-dns freeipa-server-trust-ad initscripts && dnf clean all
```
Comment (stlaz) [inline Dockerfile.fedora-rawhide]: Should this be fixed for older containers, too?

File: Dockerfile.fedora-rawhide (hunk 1 above)
Comment (adelton) [reply Dockerfile.fedora-rawhide]: Mea culpa. Fixed in a99c2a1d0f81093bb779b45ec3d7da5d08b32c53.

File: Dockerfile.fedora-rawhide (hunk 2 above)
Comment (adelton) [reply Dockerfile.fedora-rawhide]: I don't think I care. We will get the change for Fedora 28 and onwards and it does not really hard older version.

Comment (stlaz) [reply Dockerfile.fedora-rawhide]: Right, that's fine

**Comment by adelton** [comment]
Does `ipa-backup` back those up?

**Comment by flo-renaud** [comment]
Hi @adelton 
ipa-backup retrieves the current profile/options with `authselect current --raw` and saves this information in the backup, but does not directly backup the files. ipa-restore installs the profile/options with `authselect select <saved profile> [<saved options>]`.

**Comment by flo-renaud** [comment]
@adelton 
yes, I would do as you suggest, i.e. store the output of `authselect current --raw` and re-apply the same setting on subsequent start.
Re. upgrade, ipa-server-upgrade is migrating from authconfig to authselect. It is checking if the server was installed with --mkhomedir and then calls `authselect sssd [with-mkhomedir] --force`. This means that we always apply sssd profile, with optional options.

**Comment by stlaz** [comment]
We decided that we may try to go without the authselect backups since they may not necessarily affect the server functionality.

I tried to test this PR with some of our tests and it seems that a replica can't be created because CA returns CA_REJECTED while requesting DS certificate. On further investigation, it seems that CA is unable to connect to LDAP, and that's reproducible. There are some AVCs, I'll try to run this in permissive mode in the meantime and will try to investigate what could cause the AVCs.

**Comment by adelton** [comment]
Can we push the Dockerfiles so that we have content which actually builds, even if there are known issues running it? We can polish the runtime / replica bugs later but it might be easier done if we have the content out there.

**Comment by stlaz** [comment]
Sure. I guess I'd rather not yet build them on DockerHub until we get something that's a bit more stable, though, or do you think we should provide them even if (possibly) broken?

**Comment by adelton** [comment]
No need to enable building them on Docker Hub at this point.

**Comment by stlaz** [comment]
Thanks, merging.

**Pull request merged**